drivers/mfd/mfd_nct38xx.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2023 Google, LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_nct38xx
+
+#include <zephyr/drivers/i2c.h>
+
+struct mfd_nct38xx_config {
+	const struct i2c_dt_spec i2c_dev;
+};
+
+struct mfd_nct38xx_data {
+	/* lock NC38xx register access */
+	struct k_sem lock;
+};
+
+static int mfd_nct38xx_init(const struct device *dev)
+{
+	struct mfd_nct38xx_data *data = dev->data;
+
+	k_sem_init(&data->lock, 1, 1);
+
+	return 0;
+}
+
+int mfd_nct38xx_lock(const struct device *dev, k_timeout_t timeout)
+{
+	struct mfd_nct38xx_data *data = dev->data;
+
+	return k_sem_take(&data->lock, timeout);
+}
+
+void mfd_nct38xx_unlock(const struct device *dev)
+{
+	struct mfd_nct38xx_data *data = dev->data;
+
+	k_sem_give(&data->lock);
+}
+
+/**
+ * @brief Read a NCT38XX register
+ *
+ * @param dev NCT38XX multi-function device
+ * @param reg_addr Register address
+ * @param val A pointer to a buffer for the data to return
+ *
+ * @return 0 if successful, otherwise failed.
+ */
+int nct38xx_reg_read_byte(const struct device *dev, uint8_t reg_addr, uint8_t *val)
+{
+	const struct mfd_nct38xx_config *const config =
+		(const struct mfd_nct38xx_config *)dev->config;
+	return i2c_reg_read_byte_dt(&config->i2c_dev, reg_addr, val);
+}
```

Comment:
what is the point of a plain wrapper around i2c_reg_read?

Reply:
but again, what is the point of plain wrappers around i2c? just take i2c parent instance on each subdevice and use the standard i2c API (see e.g. npm6001). It would only make sense if i2c ops require extra steps/crafting like in npm1300, to guarantee atomic operations, etc.

Reply:
They are probably a leftover from before we had `_dt` macros, may make sense to do a swipe of i2c drivers and drop them where applicable, sure there's few others.

Reply:
I removed the wrapper routines, but kept the changes in a separate commit.  There are some non-trivial conversions to switch from `nct38xx_reg_update()` to `i2c_reg_update_byte_dt()`.